src/runtime_src/core/edge/drm/zocl/cu_scu.c

```diff
@@ -147,12 +152,18 @@ scu_ctrl_hs_check(struct xrt_cu_scu *scu, struct xcu_status *status, bool force)
 
 	ctrl_reg = *cu_regfile;
 	/* ap_ready and ap_done would assert at the same cycle */
-	if (ctrl_reg & CU_AP_DONE) {
+	if ((ctrl_reg == CU_AP_DONE) || scu->sk_crashed) {
```

Comment:
For PL kernel, the control register bits can be set at the same time.
It looks like in PS kernel, the content of control register is being redefined. If this is the case, the comment in before is incorrect. Please delete it.